src/script/common/c_content.cpp

```diff
@@ -334,7 +334,8 @@ void read_object_properties(lua_State *L, int index,
 	}
 	lua_pop(L, 1);
 
-	getstringfield(L, -1, "visual", prop->visual);
+	prop->visual = static_cast<ObjectVisual>(getenumfield(L, -1, "visual", es_ObjectVisual,
+		OBJECTVISUAL_UNKNOWN));
```

Comment:
This shouldn't update the value if nil.
Players are invisible btw., I guess because of this.